stats/stats.go

```diff
@@ -0,0 +1,82 @@
+// +build linux
+
+package stats
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	pb "github.com/c-fs/cfs/proto"
+	"github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/storage/memory"
+	"github.com/google/cadvisor/utils/sysfs"
+	"golang.org/x/net/context"
+)
+
+const (
+	// TODO: make container name configurable
+	//
+	// The recommended way to run cfs is putting the process in an
+	// exclusive container, which has its own cgroup in each hierarchy.
+	// It helps cfs to monitor its resource usage.
+	//
+	// It asks users to specify container name instead of detecting
+	// cgroup that has the process automatically because it may meet
+	// strange cases:
+	// 1. cfs process may be in different cpu/memory/etc cgroups
```

Comment:
@tnachen each process can belong to one cgroup in one hierachry, and it can belong to diffrent cgroup in different hierachy. For example, "cfs" process can belong to "cfs" memory cgroup and "cfs2" cpu cgroup.

I hope the user can take control of this part: he gives us a cgroup name that we could use to monitor the resource usage of cfs process. Considering we do have assumption that cfs should run in container, i think that is reasonable.